### DISCUSSION TITLE
Adoption of lint rules in Dart

### DISCUSSION THREAD

**Discussion by marcotrumpet** [eopeter/flutter_mapbox_navigation#239] [Ideas]
I want to add a [lint rule](https://pub.dev/packages/very_good_analysis) package to have some common rules about how to write code. 

This can lead to clean code helping new contributors to work on the package. 

Of course some refactoring will be needed and it will be cool to check that no warnings are displayed before accepting or submitting a PR (clean problem tab on VsCode as an example)

**Comment by eopeter** [comment]
What linting package do you propose?

**Reply by marcotrumpet** [reply]
The one I linked in the message above.. it's pretty strict but we can fine tune rules https://pub.dev/packages/very_good_analysis

**Reply by marcotrumpet** [reply]
it would be cool also to automatically reject PRs that have issues or lint problems maybe with a github action

**Comment by eopeter** [comment]
I like it. Lets do it

**Reply by marcotrumpet** [reply]
if you want I can start work on it.. hopefully there will be a lot of changes no one with impact on functionalities

**Reply by eopeter** [reply]
Please do; I will begin to look into the GitHub action

**Reply by marcotrumpet** [reply]
instant regret.. there are 427 errors.. most of the on types.. let's take the time it needs

**Comment by marcotrumpet** [comment]
added PR https://github.com/eopeter/flutter_mapbox_navigation/pull/251

**Comment by marcotrumpet** [comment]
Hey @eopeter since none of the PR is passing the action check isn't it better to remove the action or do a simple one that just runs flutter analyze command?

**Reply by eopeter** [reply]
I like the linter in General and would need to work to get it turned back on but for now, I turned off the Github Action for it. Thanks for your work on `very_good_analysis`
